sdks/java/io/mongodb/src/main/java/org/apache/beam/sdk/io/mongodb/MongoDbIO.java

```diff
@@ -519,14 +507,24 @@ public void createMongoClient() throws Exception {
         client = new MongoClient(new MongoClientURI(uri));
       }
 
+      @StartBundle
+      public void startBundle(Context ctx) throws Exception {
+        batch = new ArrayList<>();
+      }
+
       @ProcessElement
       public void processElement(ProcessContext ctx) throws Exception {
         String value = ctx.element();
 
+        batch.add(Document.parse(ctx.element()));
```

Comment:
This will OOM if you have a bundle with a lot of elements (more likely in a batch pipeline, but can happen in a streaming pipeline too). You need to limit the size of the batch and flush it both when it reaches the limit and in finishBundle().